File: api/_lib/intake-test.js
/** Detect probe/E2E test intakes — never treat as customer leads. */

const TEST_FORM_IDS = new Set(["nf_intake_e2e", "nf_probe_cron"]);
const TEST_TOPICS = new Set(["e2e", "probe"]);
const TEST_EMAILS = new Set(["e2e@noetfield.com", "probe@noetfield.com"]);

function meta(body) {
  return body && typeof body.metadata === "object" && body.metadata ? body.metadata : {};
}

function isTestIntake(body) {
  if (!body || typeof body !== "object") return false;
  const m = meta(body);
  if (m.intake_kind === "test") return true;
  const formId = String(m.form_id || "").toLowerCase();
  if (TEST_FORM_IDS.has(formId)) return true;
  const topic = String(m.topic || "").toLowerCase();
  if (TEST_TOPICS.has(topic)) return true;
  const email = String(body.contact_email || "").trim().toLowerCase();
  if (TEST_EMAILS.has(email)) return true;
  const rid = String(body.request_id || "").toUpperCase();
  if (/^RID-(E2E|PROBE)-/.test(rid)) return true;
  return false;
}

function testPipelineLabel(body) {
  const m = meta(body);
  if (m.pipeline) return String(m.pipeline);
  const formId = String(m.form_id || "").toLowerCase();
  if (formId === "nf_probe_cron") return "probe_cron:intake_e2e";
  if (formId === "nf_intake_e2e") return "nf_intake_e2e:deploy_verify";
  if (m.topic === "probe") return "probe:intake";
  if (m.topic === "e2e") return "e2e:intake";
  return "test:intake";
}

function ensureTestMetadata(body) {
  if (!isTestIntake(body)) return body;
  const m = { ...meta(body), intake_kind: "test" };
  if (!m.pipeline) m.pipeline = testPipelineLabel(body);
  return { ...body, metadata: m };
}

module.exports = {
  isTestIntake,
  testPipelineLabel,
  ensureTestMetadata,
};

File: services/governance/noetfield_governance/intake_test.py
"""Probe/E2E test intake detection — separate from customer leads."""

from __future__ import annotations

import re
from typing import Any

TEST_FORM_IDS = frozenset({"nf_intake_e2e", "nf_probe_cron"})
TEST_TOPICS = frozenset({"e2e", "probe"})
TEST_EMAILS = frozenset({"e2e@noetfield.com", "probe@noetfield.com"})
_TEST_RID_RE = re.compile(r"^RID-(E2E|PROBE)-", re.IGNORECASE)


def is_test_intake(
    *,
    metadata: dict[str, Any] | None = None,
    request_id: str | None = None,
    contact_email: str | None = None,
) -> bool:
    meta = metadata or {}
    if meta.get("intake_kind") == "test":
        return True
    if str(meta.get("form_id") or "").lower() in TEST_FORM_IDS:
        return True
    if str(meta.get("topic") or "").lower() in TEST_TOPICS:
        return True
    email = (contact_email or "").strip().lower()
    if email in TEST_EMAILS:
        return True
    rid = (request_id or "").strip().upper()
    if rid and _TEST_RID_RE.match(rid):
        return True
    return False


def pipeline_label(metadata: dict[str, Any]) -> str:
    if metadata.get("pipeline"):
        return str(metadata["pipeline"])
    form_id = str(metadata.get("form_id") or "").lower()
    if form_id == "nf_probe_cron":
        return "probe_cron:intake_e2e"
    if form_id == "nf_intake_e2e":
        return "nf_intake_e2e:deploy_verify"
    topic = str(metadata.get("topic") or "").lower()
    if topic == "probe":
        return "probe:intake"
    if topic == "e2e":
        return "e2e:intake"
    return "test:intake"


def ensure_test_metadata(
    metadata: dict[str, Any] | None,
    *,
    request_id: str | None = None,
    contact_email: str | None = None,
) -> dict[str, Any]:
    meta = dict(metadata or {})
    if not is_test_intake(metadata=meta, request_id=request_id, contact_email=contact_email):
        return meta
    meta["intake_kind"] = "test"
    if not meta.get("pipeline"):
        meta["pipeline"] = pipeline_label(meta)
    return meta
